### PR TITLE
Untested tps counter

### PR DESCRIPTION
**Pull request by buthed010203**
I dont know if its gonna work but i hope so?

### CODE REVIEW

**Comment by USMP-lancer** [comment]
only one way to find out if it works lol

**Review by USMP-lancer — APPROVED** [review]
g

File: global.js [hunk 1]
```diff
@@ -161,7 +161,22 @@ if (!ts.eventsRegistered) {
...
     // Events.on(EventType.PlayerLeave, cons(e => teamKeeper(e.player, true)))
-
+    
+    var lastTick = 0
+    var mean = new WindowedMean(60)
+    const timer = new Interval()
+    Events.run(Trigger.update, () -> {
+        mean.add(Time.sinceMillis(lastTick))
+        lastTick = Time.millis()
+        if (timer.get(60) { // Run once a second
```
Comment (Blockzilla101) [inline global.js]:
```suggestion
        if (timer.get(60)) { // Run once a second
```

File: global.js [hunk 2]
```diff
@@ -161,7 +161,22 @@ if (!ts.eventsRegistered) {
 
     // Events.on(EventType.PlayerJoin, cons(e => teamKeeper(e.player, false)))
     // Events.on(EventType.PlayerLeave, cons(e => teamKeeper(e.player, true)))
-
+    
+    var lastTick = 0
+    var mean = new WindowedMean(60)
+    const timer = new Interval()
+    Events.run(Trigger.update, () -> {
```
Comment (Blockzilla101) [inline global.js]:
```suggestion
    Events.run(Trigger.update, () => {
```

File: global.js [hunk 3]
```diff
@@ -161,7 +161,22 @@ if (!ts.eventsRegistered) {
...
+        mean.add(Time.sinceMillis(lastTick))
```
Comment (Blockzilla101) [inline global.js]: this crashes the server
image

**Review by Blockzilla101 — CHANGES_REQUESTED** [review]

File: global.js (hunk 3 above)
Comment (USMP-lancer) [reply global.js]: oh rip

**Comment by buthed010203** [comment]
Ok my brain is very incredibly small but i fixed it all

File: global.js [hunk 4]
```diff
@@ -161,7 +161,22 @@ if (!ts.eventsRegistered) {
...
+    const timer = new Interval()
+    Events.run(Trigger.update, () => {
+        mean.add(Time.timeSinceMillis(lastTick))
+        lastTick = Time.millis()
+        if (timer.get(60)) { // Run once a second
+            const tps = 1000 / mean.rawMean()
+            Groups.player.each(cons(p => {
+                Call.infoPopup(p.con, `TPS: ${tps}`, 1,  Align.top, p.con.mobile ? 0 : 120, 0, 0, 0) // 3 times to approximately match the background color of the item hud
+                Call.infoPopup(p.con, `TPS: ${tps}`, 1,  Align.top, p.con.mobile ? 0 : 120, 0, 0, 0)
+                Call.infoPopup(p.con, `TPS: ${tps}`, 1,  Align.top, p.con.mobile ? 0 : 120, 0, 0, 0)
```
Comment (Blockzilla101) [inline global.js]:
```suggestion
                Call.infoPopup(p.con, 'TPS: ' + tps, 1,  Align.top, p.con.mobile ? 0 : 120, 0, 0, 0) // 3 times to approximately match the background color of the item hud
                Call.infoPopup(p.con, 'TPS: ' + tps, 1,  Align.top, p.con.mobile ? 0 : 120, 0, 0, 0)
                Call.infoPopup(p.con, 'TPS: ' + tps, 1,  Align.top, p.con.mobile ? 0 : 120, 0, 0, 0)
```
rhino doesnt support those
image

File: global.js [hunk 5]
```diff
@@ -161,7 +161,22 @@ if (!ts.eventsRegistered) {
...
-
+    
+    var lastTick = 0
+    var mean = new WindowedMean(60)
+    const timer = new Interval()
+    Events.run(Trigger.update, () => {
+        mean.add(Time.timeSinceMillis(lastTick))
+        lastTick = Time.millis()
+        if (timer.get(60)) { // Run once a second
+            const tps = 1000 / mean.rawMean()
```
Comment (Blockzilla101) [inline global.js]:
```suggestion
            const tps = Mathf.round(1000 / mean.rawMean())
```

**Review by Blockzilla101 — CHANGES_REQUESTED** [review]

**Review by Blockzilla101 — APPROVED** [review]

**Pull request closed without merging**